src/modules/text.cpp

```diff
@@ -8,18 +9,13 @@ namespace modules {
   template class module<text_module>;
 
   text_module::text_module(const bar_settings& bar, string name_) : static_module<text_module>(bar, move(name_)) {
-    m_formatter->add("content", "", {});
-
-    if (m_formatter->get("content")->value.empty()) {
-      throw module_error(name() + ".content is empty or undefined");
```

Comment:
This looks like very old code. No other module actually does that anymore. I don't think this needs to be adapted and can be removed. What all the other modules do, is to load the format, as you did, and if the label tag occurs in the format string, load the label. For example the script modules does it like this:
```cpp
modules/script.cpp:
-------------------
m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL});
if (m_formatter->has(TAG_LABEL)) {
  m_label = load_optional_label(m_conf, name(), "label", "%output%");
}
```

This avoids throwing unnecessary errors because one should be able to also define a format without including any tags (even if it's not that useful).

Reply:
In your last example

```
if (m_formatter->has(TAG_LABEL))
```

I'm unsure about removing the `DEFAULT_FORMAT` argument, as in:

```
if (m_formatter->has(TAG_CONTENT, DEFAULT_FORMAT))
```

Because the formatter could contain more than one format and the only API I found to check if a tag is inside a format is to request the formatter for that tag in one of its formats. But maybe the dafault format is considered when no other is explicitly passed?

Reply:
From looking at the code (`base.cpp:143`), if you pass only one argument to `module_formatter::has`, then it searches through all the defined formats. So in this case it shouldn't matter, because the text module only defines a single format `format`.

Reply:
Ok, I've already updated and rebased everything according to your comments.